File: output/gemma4-e4b/avatar/Java/Python/codeforces_190_A.py
import sys

def solve():
    # Read all input data and split it into tokens
    data = sys.stdin.read().split()
    if not data:
        return

    try:
        n = int(data[0])
        m = int(data[1])
    except IndexError:
        # Handle case where not enough input is provided
        return

    if n == 0:
        if m != 0:
            print("Impossible")
        else:
            # Equivalent to Java's System.out.println("0 0")
            print("0 0")
        return

    if m <= n:
        if m == 0:
            m = 1
        # Output: n and (m + n - 1)
        print(f"{n} {m + n - 1}")
        return

    # Case m > n
    # Output: m and (m + n - 1)
    print(f"{m} {m + n - 1}")

if __name__ == "__main__":
    solve()

File: output/gemma4-e4b/avatar/Python/Java/codeforces_190_A.java
import java.util.Scanner;

public class codeforces_190_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read n and m from a single line input
        if (!scanner.hasNextInt()) {
            // Handle case where input might be empty or malformed
            return;
        }
        
        int n = scanner.nextInt();
        int m = scanner.nextInt();
        
        // Block 1: if n < 1
        if (n < 1) {
            if (m < 1) {
                System.out.println(n + " " + m);
                return; // quit()
            } else {
                System.out.println("Impossible");
                return; // quit()
            }
        }
        
        // Block 2: if m > n
        if (m > n) {
            int min = m;
            int max = n + m - 1;
            System.out.println(min + " " + max);
        }
        
        // Block 3: if m <= n
        if (m <= n) {
            int min = n;
            int max = n + m - 1;
            if (m == 0) {
                max = n;
            }
            System.out.println(min + " " + max);
        }
        
        scanner.close();
    }
}